libraries/standard/mqtt/src/mqtt_lightweight.c

```diff
@@ -2046,35 +2106,17 @@ MQTTStatus_t MQTT_GetIncomingPacketTypeAndLength( MQTTTransportRecvFunc_t readFu
                                                   MQTTPacketInfo_t * pIncomingPacket )
 {
     MQTTStatus_t status = MQTTSuccess;
-    /* Read a single byte. */
-    int32_t bytesReceived = readFunc( networkContext, &( pIncomingPacket->type ), 1U );
 
-    if( bytesReceived == 1 )
+    if( pIncomingPacket == NULL )
     {
-        /* Check validity. */
-        if( incomingPacketValid( pIncomingPacket->type ) == true )
-        {
-            pIncomingPacket->remainingLength = getRemainingLength( readFunc, networkContext );
-
-            if( pIncomingPacket->remainingLength == MQTT_REMAINING_LENGTH_INVALID )
-            {
-                status = MQTTBadResponse;
-            }
-        }
-        else
-        {
-            LogError( ( "Incoming packet invalid: Packet type=%u",
-                        pIncomingPacket->type ) );
-            status = MQTTBadResponse;
-        }
-    }
-    else if( bytesReceived == 0 )
```

Comment:
I would suggest having this be something like `else if( ( bytesReceived == 0 ) && ( status != MQTTBadParameter ) )`, so that we don't have to add another helper function.